README.rst

```diff
@@ -15,125 +15,137 @@ About
 =====
 
 This repository provides a simple "hello world" application example for `REANA
-<http://reanahub.io/>`_ reusable research data analysis plaftorm.
+<http://reanahub.io/>`_ reusable research data analysis plaftorm. The example
+generates greetings for persons specified in an input file.
 
-Code
-====
+Making a research data analysis reproducible means to provide "runnable recipes"
+addressing (1) where the input datasets are, (2) what software was used to
+analyse the data, (3) which computing environment was used to run the software,
+and (4) which workflow steps were taken to run the analysis.
 
-A simple "hello world" Python code for illustration:
-
-- `helloworld.py <helloworld.py>`_
+1. Input dataset
+================
 
-The program takes several optional arguments such as the name of the person to
-greet and an optional sleeptime period and produces an output file with the
-greeting:
+The input file is a text file containing person names:
 
 .. code-block:: console
 
-   $ python helloworld.py --name Jane --sleeptime 0
-   $ cat output/greetings.txt
-   Hello Jane!
+   $ cat inputs/names.txt
+   John Doe
+   Jane Doe
 
-Containerisation
+2. Analysis code
 ================
 
-In order to be able to rerun the code even several years in the future, we need
-to "encapsulate the current environment" for example to freeze the Python
-version the code is using. We shall achieve this by preparing a `Docker
-<https://www.docker.com/>`_ container image for the program:
+A simple "hello world" Python code for illustration:
 
-- `Dockerfile <Dockerfile>`_
+- `helloworld.py <code/helloworld.py>`_
 
-For example:
+The program takes an input file with the names of the persons to greet and an
+optional sleeptime period and produces an output file with the greetings:
 
 .. code-block:: console
 
-    $ cat Dockerfile
-    FROM python:2.7-alpine
-    ADD . /code
-    WORKDIR /code
-    CMD ["python", "helloworld.py", \
-         "--name", "John Doe", \
-         "--outputfile", "output/greetings.txt", \
-         "--sleeptime", "1"]
-
-We build the container image:
+   $ python code/helloworld.py \
+       --inputfile inputs/names.txt
+       --outputfile outputs/greetings.txt
+       --sleeptime 0
+   $ cat outputs/greetings.txt
+   Hello John Doe!
+   Hello Jane Doe!
 
-.. code-block:: console
+3. Compute environment
+======================
 
-    $ docker build -t helloworld .
+In order to be able to rerun the code even several years in the future, we need
+to "encapsulate the current environment", for example to freeze the Python
+version the code is using. We shall achieve this by preparing a `Docker
+<https://www.docker.com/>`_ container image for the program.
 
-and test whether it works locally:
+For example:
 
 .. code-block:: console
 
-    $ docker run -v `pwd`:/code -i -t --rm helloworld python helloworld.py --name Jim --sleeptime 1
-    $ tail -1 output/greetings.txt
-    Hello Jim!
+    $ cat environment/Dockerfile
+    FROM python:2.7
```

Comment:
As an example of a dependency in an additional Python package we could use a library that makes the `helloworld.py` as small as possible, such as `click` so we can profit of removing boilerplate code.. but not for now.